pallets/identity/src/lib.rs

```diff
@@ -2049,6 +2049,77 @@ impl<T: Trait> Module<T> {
             RpcDidRecords::IdNotFound
         }
     }
+
+    /// Use to get the filtered link data for a given signatory
+    /// - if link_type is None then return links data on the basis of the `allow_expired` boolean
+    /// - if link_type is Some(value) then return filtered links on the value basis type in conjunction
+    ///   with `allow_expired` boolean condition
+    pub fn get_filtered_links(
+        signatory: Signatory,
+        allow_expired: bool,
+        link_type: Option<LinkType>,
+    ) -> Vec<Link<T::Moment>> {
+        let now = <pallet_timestamp::Module<T>>::get();
+
+        if let Some(type_of_link) = link_type {
+            <Links<T>>::iter_prefix(signatory)
+                .filter(|link| {
+                    if !allow_expired {
+                        if let Some(expiry) = link.expiry {
+                            if expiry < now {
+                                return false;
+                            }
+                        }
+                    }
+                    match link.link_data {
+                        LinkData::DocumentOwned(..) => {
+                            if type_of_link == LinkType::DocumentOwnership {
+                                true
+                            } else {
+                                false
+                            }
+                        }
+                        LinkData::TickerOwned(..) => {
+                            if type_of_link == LinkType::TickerOwnership {
+                                true
+                            } else {
+                                false
+                            }
+                        }
+                        LinkData::AssetOwned(..) => {
+                            if type_of_link == LinkType::AssetOwnership {
+                                true
+                            } else {
+                                false
+                            }
+                        }
+                        LinkData::NoData => {
+                            if type_of_link == LinkType::NoData {
+                                true
+                            } else {
+                                false
+                            }
+                        }
+                    }
+                })
+                .map(|l| l)
```

Comment:
Is this needed?

Reply:
I am guessing Satyam is trying to take ownership of the element here. There are other ways to do it like cloning but this works as well :)